grpc/execution/server_test.go

```diff
@@ -0,0 +1,463 @@
+
+				chainDestinationAddress := crypto.PubkeyToAddress(chainDestinationAddressPrivKey.PublicKey)
+
+				depositTx := &sequencerblockv1alpha1.RollupData{Value: &sequencerblockv1alpha1.RollupData_Deposit{Deposit: &sequencerblockv1alpha1.Deposit{
+					BridgeAddress: &primitivev1.Address{
+						Inner: bridgeAddress,
+					},
+					AssetId:                 bridgeAssetDenom[:],
+					Amount:                  depositAmount,
+					RollupId:                &primitivev1.RollupId{Inner: genesisInfo.RollupId},
+					DestinationChainAddress: chainDestinationAddress.String(),
+				}}}
+
+				marshalledTxs = append(marshalledTxs, depositTx)
+			}
+
+			executeBlockReq := &astriaPb.ExecuteBlockRequest{
+				PrevBlockHash: tt.prevBlockHash,
+				Timestamp: &timestamppb.Timestamp{
+					Seconds: int64(tt.timestamp),
+				},
+				Transactions: marshalledTxs,
+			}
+
+			executeBlockRes, err := serviceV1Alpha1.ExecuteBlock(context.Background(), executeBlockReq)
+			if tt.expectedReturnCode > 0 {
+				require.NotNil(t, err, "ExecuteBlock should return an error")
+				require.Equal(t, tt.expectedReturnCode, status.Code(err), "ExecuteBlock failed: %v", err)
+			}
+			if err == nil {
+				require.NotNil(t, executeBlockRes, "ExecuteBlock response is nil")
+
+				astriaOrdered := ethservice.TxPool().AstriaOrdered()
+				require.Equal(t, 0, astriaOrdered.Len(), "AstriaOrdered should be empty")
+
+				// check if commitment state is not updated
+				commitmentStateAfterExecuteBlock, err := serviceV1Alpha1.GetCommitmentState(context.Background(), &astriaPb.GetCommitmentStateRequest{})
+				require.Nil(t, err, "GetCommitmentState failed: %v", err)
+
+				require.Exactly(t, commitmentStateBeforeExecuteBlock, commitmentStateAfterExecuteBlock, "Commitment state should not be updated")
+			}
+
+		})
+	}
+}
+
+func TestExecutionServiceServerV1Alpha2_ExecuteBlockAndUpdateCommitment(t *testing.T) {
+	ethservice, serviceV1Alpha1 := setupExecutionService(t, 10)
+
+	// call genesis info
+	genesisInfo, err := serviceV1Alpha1.GetGenesisInfo(context.Background(), &astriaPb.GetGenesisInfoRequest{})
+	require.Nil(t, err, "GetGenesisInfo failed: %v", err)
+	require.NotNil(t, genesisInfo, "GenesisInfo is nil")
+
+	// call get commitment state
+	commitmentState, err := serviceV1Alpha1.GetCommitmentState(context.Background(), &astriaPb.GetCommitmentStateRequest{})
+	require.Nil(t, err, "GetCommitmentState failed: %v", err)
+	require.NotNil(t, commitmentState, "CommitmentState is nil")
+
+	// get previous block hash
+	previousBlock := ethservice.BlockChain().CurrentSafeBlock()
+	require.NotNil(t, previousBlock, "Previous block not found")
+
+	// create 5 txs
+	txs := []*types.Transaction{}
+	marshalledTxs := []*sequencerblockv1alpha1.RollupData{}
+	for i := 0; i < 5; i++ {
+		unsignedTx := types.NewTransaction(uint64(i), common.HexToAddress("0x9a9070028361F7AAbeB3f2F2Dc07F82C4a98A02a"), big.NewInt(1), params.TxGas, big.NewInt(params.InitialBaseFee*2), nil)
```

Comment:
same here about the address

Reply:
actually make it a global var since it seems to be used a lot of places

Reply:
done!